solidity/contracts/converter/BancorConverterUpgrader.sol

```diff
@@ -230,9 +237,15 @@ contract BancorConverterUpgrader is IBancorConverterUpgrader, ContractRegistryCl
 
         for (uint16 i = 0; i < connectorTokenCount; i++) {
             address connectorAddress = _oldConverter.connectorTokens(i);
-            IERC20Token connector = IERC20Token(connectorAddress);
-            connectorBalance = connector.balanceOf(_oldConverter);
-            _oldConverter.withdrawTokens(connector, address(_newConverter), connectorBalance);
+            if (connectorAddress == address(0)) {
+                connectorBalance = etherToken.balanceOf(_oldConverter);

```

Comment:
If the connector address is 0, it means there's no EtherToken involved anymore

Reply:
@yudilevi : Note the flow:
1. In `copyConnectors`, we need to check `if (connectorAddress == address(etherToken))`. I've extended it to `if (connectorAddress == address(0) || connectorAddress == address(etherToken))`, so that it will be usable also for upgrading our new ETH converter in the future. Otherwise, we'd have to redeploy the upgrader in order to do that.
2. In `transferConnectorsBalances`, we've already added a zero reserve representing ETH, so we just need to withdraw the funds from Ether token into the converter.